src/test/ProposalPayloadE2E.t.sol

```diff
@@ -1,39 +1,386 @@
+        assertEq(CRV.balanceOf(AaveV2Ethereum.COLLECTOR), initialCollectorCrvBalance + amount);
+        // Compensating for +1/-1 precision issues when rounding, mainly on aTokens
+        assertApproxEqAbs(AUSDC.balanceOf(CRV_WHALE), initialPurchaserAusdcBalance + ausdcAmountOut, 1);
+        assertEq(CRV.balanceOf(CRV_WHALE), initialPurchaserCrvBalance - amount);
+
+        assertEq(crvRepayment.totalAUSDCSold(), ausdcAmountOut);
+        assertEq(crvRepayment.totalCRVReceived(), amount);
+    }
+
+    function testPurchaseWithdrawFromAaveTrueFuzz(uint256 amount) public {
+        // Pass vote and execute proposal
+        GovHelpers.passVoteAndExecute(vm, proposalId);
+
+        // Assuming upper bound of purchase of 2,650,000 CRV and lower bound of 2 CRV (price is less than 1)
+        vm.assume(amount >= 2e18 && amount <= crvRepayment.CRV_CAP());
+
+        vm.startPrank(CRV_WHALE);
+        CRV.approve(address(crvRepayment), amount);
+
+        uint256 initialCollectorAusdcBalance = AUSDC.balanceOf(AaveV2Ethereum.COLLECTOR);
+        uint256 initialCollectorCrvBalance = CRV.balanceOf(AaveV2Ethereum.COLLECTOR);
+        uint256 initialPurchaserUsdcBalance = USDC.balanceOf(CRV_WHALE);
+        uint256 initialPurchaserCrvBalance = CRV.balanceOf(CRV_WHALE);
+
+        assertEq(crvRepayment.totalAUSDCSold(), 0);
+        assertEq(crvRepayment.totalCRVReceived(), 0);
+
+        uint256 usdcAmountOut = crvRepayment.purchase(amount, true);
+
+        // Aave V2 Collector gets some additional aTokens minted to it due to withdrawal happening in the purchase() function
+        // see: https://github.com/aave/protocol-v2/blob/baeb455fad42d3160d571bd8d3a795948b72dd85/contracts/protocol/libraries/logic/ReserveLogic.sol#L265-L325
+        assertApproxEqAbs(AUSDC.balanceOf(AaveV2Ethereum.COLLECTOR), initialCollectorAusdcBalance - usdcAmountOut, 1);
+        assertEq(CRV.balanceOf(AaveV2Ethereum.COLLECTOR), initialCollectorCrvBalance + amount);
+        assertEq(USDC.balanceOf(CRV_WHALE), initialPurchaserUsdcBalance + usdcAmountOut);
+        assertEq(CRV.balanceOf(CRV_WHALE), initialPurchaserCrvBalance - amount);
+
+        // Compensating for +1/-1 precision issues when rounding while transferring aTokens in the purchase() function
+        assertApproxEqAbs(crvRepayment.totalAUSDCSold(), usdcAmountOut, 1);
+        assertEq(crvRepayment.totalCRVReceived(), amount);
+    }
+
+    function testInvalidPriceFromOracleFuzz(int256 price) public {
+        vm.assume(price <= int256(0));
+
+        // Mocking returned value of price <=0
+        vm.mockCall(
+            address(CRV_USD_FEED),
+            abi.encodeWithSelector(CRV_USD_FEED.latestRoundData.selector),
+            abi.encode(uint80(10), price, uint256(2), uint256(3), uint80(10))
+        );
+
+        vm.expectRevert(CRVBadDebtRepayment.InvalidOracleAnswer.selector);
+        crvRepayment.getOraclePrice();
+
+        vm.clearMockedCalls();
+    }
+
+    function testRepaySuccessful() public {
```

Comment:
A test should be added if a fix is added on the repay method, to repay with not full amount